### PR TITLE
Fix count for non-existant column

### PR DESCRIPTION
**Pull request by pustovit**

### CODE REVIEW

**Comment by dobe** [comment]
pls update CHANGES.txt

File: sql/src/main/java/io/crate/executor/transport/task/elasticsearch/ESQueryBuilder.java [hunk 1]
```diff
@@ -118,13 +118,19 @@ public static String convertWildcardToRegex(String wildcardString) {
...
+            context.builder.field(Fields.QUERY, "");
+            context.builder.endObject(); // query_string
+            context.builder.endObject(); // query
+            return;
+        }
+        context.builder.startObject(Fields.QUERY);
+        if (whereClause.hasQuery()) {
             visitor.process(whereClause.query(), context);
-        } else {
+        } else if(!whereClause.noMatch()) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/executor/transport/task/elasticsearch/ESQueryBuilder.java]: remove condition, its redundant

File: sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java [hunk 2]
```diff
@@ -116,6 +116,17 @@ public void testSelectCountStar() throws Exception {
     }
 
     @Test
+    public void testSelectCountStarWithWhereClauseForUnknownCol() throws Exception {
+        execute("create table test (\"type\" string) with (number_of_replicas=0)");
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java]: please rename type column to name

**Comment by mfussenegger** [comment]
ack

**Pull request merged**